sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -281,12 +346,6 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
       // --- Without joining keys ------------------------------------------------------------
 
       // Pick BroadcastNestedLoopJoin if one side could be broadcast
-      case j @ logical.Join(left, right, joinType, condition, hint)
-          if canBroadcastByHints(joinType, left, right, hint) =>
```

Comment:
Moving this will change the existing join strategy selection, right?

Reply:
No, it is the same logic:
1) equi-join with "doable" broadcast hint -> broadcast-hash-join
2) equi-join without "doable" broadcast hint -> either shuffle-hash or smj
3) non-equi-join with "doable" broadcast hint -> broadcast-nested-loop join
4) non-equi-join without "doable" broadcast hint -> cartesian-product or broadcast-nested-loop